src/pages/[platform]/ai/conversation/connect-your-frontend/index.mdx

```diff
@@ -0,0 +1,451 @@
+Supported image formats are `png`, `gif`, `jpeg`, and `webp`.
+
+```ts
+const { data: message, errors } = await chat.sendMessage({
+  content: [
+    {
+      image: {
+        format: 'png',
+        source: {
+          bytes: new Uint8Array([1, 2, 3]),
+        },
+      },
+    },
+  ],
+});
+```
+
+Mixing `text` and `image` in a single message is supported.
+
+```ts
+const { data: message, errors } = await chat.sendMessage({
+  content: [
+    {
+      text: 'describe the image in detail',
+    },
+    {
+      image: {
+        format: 'png',
+        source: {
+          bytes: new Uint8Array([1, 2, 3]),
+        },
+      },
+    },
+  ],
+});
+```
+
+#### AI context
+
+The `aiContext` argument allows you to optionally attach arbitrary data to the message. This is useful for passing additional information, like user information or current state of your application, in a user message to the AI assistant.
+
+```ts
+const { data: message, errors } = await chat.sendMessage('Hello, world!', {
+  aiContext: {
+    user: {
+      name: 'Ian',
+    },
+  },
+});
+```
+
+#### Tool Configuration
+
+The `toolConfiguration` argument allows you to optionally pass a client tool configuration to the AI assistant with a user message. See the [Tools concept page](/ai/concepts/tools) and [Tools guide](/ai/conversation/tools/) for more information on how tools works.
+
+<Callout type="info">
+  Client tools are conceptually the same as data tools and lambda executable tools. They are API definitions provided to an LLM alongside a user message. The LLM can use the provided tool configuration to decide which tool (if any) to call in order to better respond to the user. However, there's an important distinction with client tools -- you are responsible for implementing the tool execution logic and responding to the AI assistant with the tool's response.
+</Callout>
+
+The `json` property is simply a JSON Schema definition of the tool's input. The AI assistant will use this schema to provide the expected input to your tool.
+```ts
+const { data: message, errors } = await conversation.sendMessage({
+  content: [
+    {
+      text: "I'd like to make a chocolate cake for my friend with a gluten intolerance. What ingredients do I need?",
+    },
+  ],
+  toolConfiguration: {
+    tools: {
+      generateRecipe: {
+        description: "List ingredients needed for a recipe",
+        inputSchema: {
+          json: {
+            type: "object",
+            properties: {
+              ingredients: {
+                type: "array",
+                items: { type: "string" },
+              },
+            },
+          },
+        },
+      },
+    },
+  },
+});
+```
+
+The response from the AI assistant will be a JSON object that matches the `inputSchema` definition. See [Subscribe to assistant responses](#subscribe-to-assistant-responses) for more information on how to handle the response.
+
+### Subscribe to assistant responses
+
```

Comment:
We could add a callout here that the React hook takes care of reconciling the stream events into, just to not scare people thinking they have to do this stuff

Reply:
Agreed. I didn't add it in this PR because the examples are only for the data client. We should include this once we have good hook examples.